cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/instantsearch-config.js

```diff
@@ -720,50 +730,121 @@ function getPriceHtml(product) {
+        const { promotionalPriceFromPromos, msg } = applyPromotions(item, algoliaData, activeCustomerPromos);
+        if (promotionalPriceFromPromos < displayPrice) {
+            displayPrice = promotionalPriceFromPromos;
+            calloutMsg = msg;
         }
+    }
+
+    if (!strikeoutPrice) {
+        strikeoutPrice = item.price > displayPrice ? item.price : null;
+    }
 
-        return {
-            price: minPrice,
-            calloutMsg: calloutMsg,
+    return { displayPrice, strikeoutPrice, calloutMsg };
+}
+
+/**
+ * Apply promotions to find a lower promotional price if available
+ * @param {Object} item The item
+ * @param {Object} algoliaData The Algolia data config
+ * @param {Array} activeCustomerPromotions Active promotions
+ * @returns {Object} { promotionalPriceFromPromos, msg }
+ */
+function applyPromotions(item, algoliaData, activeCustomerPromos) {
+    let promotionalPriceFromPromos = item.price;
+    let msg = '';
+
+    const productPromos = item.promotions[algoliaData.currencyCode];
+    for (var i = 0; i < activeCustomerPromos.length; i++) {
+        for (var j = 0; j < productPromos.length; j++) {
+            if (productPromos[j].promoId === activeCustomerPromos[i].id && productPromos[j].price < promotionalPriceFromPromos) {
+                promotionalPriceFromPromos = productPromos[j].price;
+                msg = activeCustomerPromos[i].calloutMsg;
+            }
         }
     }
+    return { promotionalPriceFromPromos, msg };
+}
 
-    if (variant.promotionalPrice) {
-        return {
-            price: variant.promotionalPrice[algoliaData.currencyCode] || variant.promotionalPrice,
-            calloutMsg: '',
+/**
+ * Apply pricebooks to adjust prices if no promotional price was found
+ * Only consider pricebooks that are currently online
+ * @param {Object} item The item
+ * @param {Object} algoliaData The Algolia data config
+ * @returns {Object} { computedPrice, computedStrikeout }
+ */
+function applyPricebooks(item, algoliaData) {
+    const validPricebooks = item.pricebooks[algoliaData.currencyCode].filter(pricebook => {
+        if (pricebook.onlineFrom && pricebook.onlineFrom > Date.now()) {
+            return false;
+        }
+        if (pricebook.onlineTo && pricebook.onlineTo < Date.now()) {
+            return false;
         }
+        return true;
+    });
+
+    if (validPricebooks.length === 0) {
+        return { computedPrice: null, computedStrikeout: null };
     }
 
-    return {
-        price: variant.price[algoliaData.currencyCode] || variant.price,
-        calloutMsg: '',
+    const prices = validPricebooks.map(pricebook => pricebook.price);
+    const maxPrice = prices.reduce((acc, currentValue) => Math.max(acc, currentValue), -Infinity);
+    const minPrice = prices.reduce((acc, currentValue) => Math.min(acc, currentValue), Infinity);
+
+    if (maxPrice > item.price) {
+        return { computedPrice: maxPrice, computedStrikeout: item.price };
```

Comment:
The logic is too error-prone:

First, this function computes a `computedStrikeout` that is never used anywhere

Then, `computedPrice` sounds like it will be the display price, but I see above that it's used as the `strikeoutPrice`
```js
        } else if (computedPrice > displayPrice) {
                // If pricebooks show a higher original price
                strikeoutPrice = computedPrice;
            }
        }
```

I think this function should simply return the max pricebooks price, which then becomes the strikeout price. Because the min pricebook price is already the price that we have in the record (the SFCC price model already returns the min pricebook price as the active price), so the condition below will never be true.